### PR TITLE
Move common gateway registration code to new package

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

Deduplication of beacon node registration in grpc-gateway.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

N/A

### CODE REVIEW

**Review by rauljordan — COMMENTED** [review]
Can we name the package to beacon-chain/gateway? Shared is weird as we use it to mean something else at the top-level

**Review by rauljordan — APPROVED** [review]

File: beacon-chain/server/main.go [hunk 1]
```diff
@@ -9,16 +9,12 @@ import (
 	"net/http"
 	"strings"
 
-	gwruntime "github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	joonix "github.com/joonix/log"
+	gateway2 "github.com/prysmaticlabs/prysm/beacon-chain/gateway"
```
Comment (terencechain) [inline beacon-chain/server/main.go:13]: Can we use a better name than gateway2?

Comment (rauljordan) [reply beacon-chain/server/main.go:13]: I think it was goland refactor lol

**Comment by rauljordan** [comment]
Ready again @terencechain

**Review by terencechain — APPROVED** [review]

**Pull request merged**